storage/innobase/arch/arch0page.cc

```diff
@@ -1716,10 +1716,11 @@ void Arch_Page_Sys::track_page(buf_page_t *bpage, lsn_t track_lsn,
   }
 
   /* We need to track this page. */
-  arch_oper_mutex_enter();
 
   while (true) {
+    arch_oper_mutex_enter();
     if (m_state != ARCH_STATE_ACTIVE) {
+      arch_oper_mutex_exit();
```

Comment:
Not an issue.

FYI, `sentry` objects can be used in such cases to ensure that the lock is released when the control gets out of scope.

Usage is as below
```
#include "raii/sentry.h" /* raii::Sentry */
...
...
arch_oper_mutex_enter();
/* Create a sentry object to call `arch_oper_mutex_exit()` on exit */
raii::Sentry<> arch_oper_mutex_guard{[&]() { arch_oper_mutex_exit(); }};
```

Reply:
Good suggestion. no need to create dummy class for RAII